src/CoreWCF.Primitives/src/CoreWCF/Security/SecuritySessionServerSettings.cs

```diff
@@ -1956,6 +1967,562 @@ public Task OpenAsync(CancellationToken token)
+                {
+                    IncomingChannel = (IDuplexSessionChannel)channel;
+                    _serviceProvider = IncomingChannel.GetProperty<IServiceScopeFactory>().CreateScope().ServiceProvider;
+                }
+
+                public IDuplexSessionChannel IncomingChannel { get; set; }
+                public IServiceChannelDispatcher ChannelDispatcher { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
+
+                public event EventHandler Closed;
+                public event EventHandler Closing;
+                public event EventHandler Faulted;
+                public event EventHandler Opened;
+                public event EventHandler Opening;
+
+                public Task DispatchAsync(RequestContext context)
+                {
+                    return DispatchAsync(context.RequestMessage);
+                }
+
+                public Task DispatchAsync(Message message)
+                {
+                    DuplexSessionRequestContext duplexSessionRequestContext = new
+                        DuplexSessionRequestContext(IncomingChannel, message);
+                    RequestContext context =  ReceiveRequest(duplexSessionRequestContext);
+                    return _channelDispatcher.DispatchAsync(context == null ? null :context.RequestMessage);
+                }
+
+                public T GetProperty<T>() where T : class
+                {
+                    T tObj = _serviceProvider.GetService<T>();
+                    if (tObj == null)
+                    {
+                        return IncomingChannel.GetProperty<T>();
+                    }
+                    else
+                    {
+                        return tObj;
+                    }
+                }
+
+                public Task OpenAsync()
+                {
+                    return Task.CompletedTask;
+                    // NO op, call by Channel Handler in the upper chain from CreateServiceChannel dispatcher from below OpenAsync.
+                }
+
+                public override async Task OpenAsync(TimeSpan timeout)
+                {
+                    await base.OpenAsync(timeout);
+                    _channelDispatcher = await Settings.SecurityServiceDispatcher.
+                        GetInnerServiceChannelDispatcher(this);
+                }
+
+                public Task OpenAsync(CancellationToken token)
+                {
+                    return OpenAsync();
+                }
+
+                public Task SendAsync(Message message)
+                {
+                    return SendAsync(message, new TimeoutHelper(ServiceDefaults.SendTimeout).GetCancellationToken());
+                }
+
+                public Task SendAsync(Message message, CancellationToken token)
+                {
+                    //moved below code t dispatch async
+                    CheckOutputOpen();
```

Comment:
This comment isn't making a lot of sense, also there's a typo. Can you clean up the old commented out line of code too?